src/parser.rs

```diff
@@ -84,18 +84,31 @@ pub fn parse_output(tokens: &[Token]) -> Result<Output> {
         }
 
         try!(expect(&mut iter, Colon));
+        let mut comma_previous = false;
+        let mut first = true;
 
         // loops through the argument list after the filter name
         while iter.peek() != None && iter.peek().unwrap() != &&Pipe {
             match iter.next().unwrap() {
-                &Comma => continue, // next argument
+                &Comma => {
+                    comma_previous = true;
+                    continue // next argument
+                }
                 x @ &StringLiteral(_) |
                 x @ &NumberLiteral(_) |
-                x @ &BooleanLiteral(_) => args.push(try!(Value::from_token(x))),
+                x @ &BooleanLiteral(_) => args.push(VarOrVal::Val(try!(Value::from_token(x)))),
+                &Identifier(ref v) => {
+                    if !comma_previous && !first {
+                        return Error::parser("a comma or a pipe", Some(&Token::Identifier(v.clone())));
+                    }
+                    args.push(VarOrVal::Var(Variable::new(v)))
+                }
                 x => {
                     return Error::parser("a comma or a pipe", Some(x));
                 }
             }
+            comma_previous = false;
+            first = false;
```

Comment:
Mmh the original code seems generally problematic since we just swallow any value without looking at commas in between. As far as I understand the Liquid "spec" there should always be a comma or a pipe after any filter argument. 

The smartest thing to do might be to add a `peek` after the `match` that checks if we have either a comma or a pipe after our argument and fail if we don't. If there is a comma we can consume it and `continue` and if there's a Pipe we can just `break`. Then you'll obviously need to remove the Comma match clause.

Do you think that'll work?